internal/jimm/access.go

```diff
@@ -804,3 +804,59 @@ func (j *JIMM) ListGroups(ctx context.Context, user *openfga.User, pagination pa
 	}
 	return groups, nil
 }
+
+// OpenFGACleanup queries OpenFGA for all existing tuples, tries to resolve each tuple and removes those
+// that JIMM cannot resolved - orphaned tuples. JIMM not being able to resolve a tuple means that the
+// corresponding entity has been removed from JIMM's database.
+func (j *JIMM) OpenFGACleanup(ctx context.Context) error {
+	for {
+		tuples, continuationToken, err := j.OpenFGAClient.ReadRelatedObjects(ctx, openfga.Tuple{}, 10, "")
+		if err != nil {
+			zapctx.Error(ctx, "reading all tuples", zap.Error(err))
+			return err
+		}
+		zapctx.Debug(ctx, "removing orphaned tuples", zap.Any("tuples", tuples))
+		orphanedTuples := j.orphanedTuples(ctx, tuples...)
+		if len(orphanedTuples) == 0 {
```

Comment:
Seems to be the wrong way to check for termination, just check if the continuation token is empty and if so, return. Which is already being done below.

Reply:
changed/fixed